aws/resource_aws_backup_plan_test.go

```diff
@@ -165,6 +167,50 @@ func TestAccAwsBackupPlan_withLifecycle(t *testing.T) {
 	})
 }
 
+func TestAccAwsBackupPlan_withLifecycleDeleteAfterOnly(t *testing.T) {
+	var plan backup.GetBackupPlanOutput
+	rStr := "lifecycle_policy_two"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsBackupPlanDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBackupPlanWithLifecycleDeleteAfterOnly(rStr),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsBackupPlanExists("aws_backup_plan.test", &plan),
+					resource.TestCheckResourceAttr("aws_backup_plan.test", "rule.2156287050.lifecycle.#", "1"),
+					resource.TestCheckResourceAttr("aws_backup_plan.test", "rule.2156287050.lifecycle.0.delete_after", "7"),
+					resource.TestCheckResourceAttr("aws_backup_plan.test", "rule.2156287050.lifecycle.0.cold_storage_after", "0"),
```

Comment:
I don't speak Go at all so I'm not sure to understand properly what you did.
AFAICT if `MoveToColdStorageAfterDays` is not sent to the AWS API, then this field won't ever appear on the backup plan. Here is an excerpt from a backup plan :
```
                "Lifecycle": {
                    "DeleteAfterDays": 90
                },
```

that has been created with this command : 
```sh
$ aws backup update-backup-plan --backup-plan-id <id> --backup-plan '{"BackupPlanName":"my-
backup", "Rules": [{"RuleName": "dailybackups", "TargetBackupVaultName": "my-vault", "Lifecycle": {"DeleteAfterDays": 90}}]}'
```

So if I understand your tests correctly you shouldn't test if the value is zero, but just ignore the field.


The same thing is true for `DeleteAfterDays`

Reply:
OK I think I got it, it is only the TF representation of the backup plan that holds the zero values, the zero values are then filtered out when the plan is sent to the AWS API.

Reply:
> OK I think I got it, it is only the TF representation of the backup plan that holds the zero values, the zero values are then filtered out when the plan is sent to the AWS API.

Hi @dud225 thanks for the review and the question. Your understanding is correct by default the zero value of any Terraform configuration would be assigned within Terraform but we would only send the value to the API if it is a non zero value, in this case some an  int greater than 0. FYI when I say zero value I am referring to the zero values of Go types https://www.callicoder.com/golang-variables-zero-values-type-inference/#zero-values